src/Microsoft.AspNetCore.Http/HeaderDictionary.cs

```diff
@@ -195,7 +195,7 @@ public void Add(string key, StringValues value)
 
             if (Store == null)
             {
-                Store = new Dictionary<string, StringValues>(1);
+                Store = new Dictionary<string, StringValues>(1, StringComparer.OrdinalIgnoreCase);
```

Comment:
Nice. Can you take it one step further by adding an EnsureStore() method with this `if` in it? Then all of the creation code paths can call the same method.